packages/common/src/directives/ng_optimized_image/preconnect_link_checker.ts

```diff
@@ -102,7 +104,7 @@ export class PreconnectLinkChecker {
    * @param originalNgSrc ngSrc value
    */
   assertPreconnect(rewrittenSrc: string, originalNgSrc: string): void {
-    if (!this.window) return;
+    if (!this.window || this.isServer) return;
```

Comment:
I'm wondering if we should just check if the current platform is server and exit in this case:
```suggestion
    if (this.isServer) return;
```
I think that was the intention of the `window`-based check.

Reply:
I kept it because `this.window ` is typed as `Window | null`

Reply:
Thanks for additional information. In this case we can move this check to the constructor:
```
    if (!this.isServer && typeof win !== 'undefined') {
      this.window = win;
    }
```
So we set the window reference only when we are not on the server. What do you think?

Reply:
@JeanMeche, I understand you didn't originally work on this part, but there's an issue with how we're obtaining the base href, and it also highlights a couple of other issues.

1. We're currently using the Window object, which is incorrect. Instead, we should be using the `PlatformLocation` service. This service will return the correct base when running on the server. When an absolute URL is provided, the `PlatformLocation` service handles it correctly.
2. When a relative URL is provided in the server config, we're currently setting the base href to `/`, which is not valid, as you pointed out. This setting lacks a protocol and makes the `href` invalid. https://github.com/angular/angular/blob/101edda0184b458d331163b7e04280c3b8b48bfc/packages/platform-server/src/domino_adapter.ts#L24. 

Maybe the `getURL` function should be updated to use a `base` instead of a Window object:

```ts
export function getUrl(src: string, base: string): URL
```

This approach is essentially equivalent to using `new URL()`, because if `src` is absolute, `new URL(src)` yields the same result as `new URL(src, win.location.href)`.

Reply:
@alan-agius4 quick question: the change is in this PR should make this service a noop on the server (only be invoked in a browser). Would you recommend changing the mentioned logic anyway?

Reply:
I agree that accessing the window global isn't great. I've refactored to use `PlatformLocation`.